autonomic-plugin-common/src/main/java/br/com/autonomiccs/autonomic/plugin/common/services/AutonomiccsSystemVmDeploymentService.java

```diff
@@ -125,6 +132,200 @@
+     *      <li>then, try to look for a host using method {@link #searchForRandomHostInClusterToDeployAutonomiccsSystemVm(ClusterVO)}
+     *  </ul>
+     * @param pod - the Pod in which we will try to look for a host to deploy a system VM
+     * @return {@link HostVO} that is going to be used to deploy a system VM. It may be returned a null value, which indicates that we did not find a suitable host to deploy the system VM in the given pod
+     */
+    public HostVO searchForRandomHostInPodToDeployAutonomiccsSystemVm(HostPodVO pod) {
+        List<ClusterVO> allClustersFromPod = clusterService.listAllClustersFromPod(pod.getId());
+        Collections.shuffle(allClustersFromPod);
+        for (ClusterVO c : allClustersFromPod) {
+            HostVO host = searchForRandomHostInClusterToDeployAutonomiccsSystemVm(c);
+            if (host != null) {
+                return host;
+            }
+        }
+        logger.info(String.format("Could not find any suitable hosts to deploy the system VM into pod [podId=%d, podName=%s]", pod.getId(), pod.getName()));
+        return null;
+    }
+
+    /**
+     * This method tries to find a suitable host to deploy an Autonomiccs system VM.
+     * It will send an empty list as the exclude host list to {@link #searchForAnotherRandomHostInTheClusterToStartSystemVmExcludingHosts(ClusterVO, List)
+     * @see #searchForAnotherRandomHostInTheClusterToStartSystemVmExcludingHosts(ClusterVO, List)
+     *
+     * @param c cluster to look for a host to deploy a system VM
+     * @return {@link HostVO} to deploy a system VM, it may return null if no suitable hosts have been found
+     */
+    private HostVO searchForRandomHostInClusterToDeployAutonomiccsSystemVm(ClusterVO c) {
+        return searchForAnotherRandomHostInTheClusterToStartSystemVmExcludingHosts(c, new ArrayList<HostVO>());
+    }
+
+    /**
+     * This method tries to find a suitable host to deploy an Autonomiccs system VM.
+     * The process is the following:
+     * <ul>
+     *   <li>We load all hosts of the given cluster, we also shuffle the host list;
+     *   <li>after that, we go over the host list and check if it can host the system VM {@link #canHostSupportVm(ServiceOfferingVO, HostResources)}
+     *   <li>if the host has resources to support the system VM, we check if its hypervisor type has a system VM template registered for it, if so we use it as the host to deploy the system VM
+     * </ul>
+     * @param c cluster to look for a host to deploy a system VM
+     * @param excludeHosts hosts to exclude from the search
+     * @return {@link HostVO} to deploy a system VM, it may return null if no suitable hosts have been found
+     */
+    private HostVO searchForAnotherRandomHostInTheClusterToStartSystemVmExcludingHosts(ClusterVO c, List<HostVO> excludeHosts) {
+        List<HostVO> allHostsInCluster = hostService.listAllHostsInCluster(c);
+        allHostsInCluster.removeAll(allHostsInCluster);
```

Comment:
hi @rafaelweingartner.
I think that the param at line 234 is "excludeHosts" and not the actual "allHostsInCluster".